nvflare/private/fed/server/training_cmds.py

```diff
@@ -201,47 +196,39 @@ def restart(self, conn: Connection, args: List[str]):
             raise TypeError("engine must be ServerEngine but got {}".format(type(engine)))
 
         if engine.job_runner.running_jobs:
-            conn.append_error("There are still jobs running. Please let them finish or abort_job before restart.")
+            msg = "There are still jobs running. Please let them finish or abort_job before restart."
+            conn.append_error(msg, meta=make_meta(MetaStatusValue.JOB_RUNNING, msg))
             return
 
         target_type = args[1]
-        if target_type == self.TARGET_TYPE_SERVER or target_type == self.TARGET_TYPE_ALL:
-
+        if target_type in [self.TARGET_TYPE_SERVER, self.TARGET_TYPE_ALL]:
             clients = engine.get_clients()
             if clients:
                 conn.append_string("Trying to restart all clients before restarting server...")
                 tokens = [c.token for c in clients]
                 conn.set_prop(
                     self.TARGET_CLIENT_TOKENS, tokens
                 )  # need this because not set in validate_command_targets when target_type == self.TARGET_TYPE_SERVER
-                response = self._restart_clients(conn, tokens)
+                response = self._restart_clients(conn)
                 conn.append_string(response)
                 # check with Isaac - no need to wait!
                 # time.sleep(5)
 
             err = engine.restart_server()
             if err:
-                conn.append_error(err)
+                conn.append_error(err, meta={MetaKey.SERVER_STATUS: MetaStatusValue.ERROR, MetaKey.INFO: err})
             else:
-                conn.append_string("Server scheduled for restart")
+                conn.append_string("Server scheduled for restart", meta={MetaKey.SERVER_STATUS: MetaStatusValue.OK})
```

Comment:
should we also use "make_meta" method here?

Reply:
No, we can't here since make_meta is used to make the final status meta.